File: JavaScript/프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.js
function solution(arr) {
    
    const asc = [...arr].sort((a, b) => {
        return a - b;
    }).slice(0, 2).reduce((c, v) => c *= v);
    const desc = [...arr].sort((a, b) => {
        return b - a;
    }).slice(0, 2).reduce((c, v) => c *= v);
    
    // asc slice떼고 0번째 1번째 / legnth-1번째 length-2번째 비교해도 됨.
    // return asc > desc ? asc : desc;
    return Math.max(asc, desc);
    
}

File: Java/프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.java
import java.util.*;

class Solution {
    public int solution(int[] numbers) {
        
        /** T1 */
        // 오름차순 정렬 후 앞 2개, 뒤 2개 곱해서 비교
        Arrays.sort(numbers);
        // for(int a : numbers)
        //     System.out.println(a);
        int a = numbers[0] * numbers[1];
        int b = numbers[numbers.length - 2] * numbers[numbers.length - 1];
        if(a > b) return a;
        else return b;
        // return Math.max(a, b);
        
        // return Math.max(numbers[0] * numbers[1], numbers[numbers.length - 2] * numbers[numbers.length - 1]);
        
    }
}